src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilder.java

```diff
@@ -241,6 +257,40 @@ static void addMetricsLabel(FilePath manifestFile) throws InterruptedException,
     manifests.write();
   }
 
+  /**
+   * Determines the namespace to use for the deployment, adds it to all of the provided manifests
+   * and returns the namespace that will be used. If namespace parameter is "*", then the namespace
+   * used will be the namespace provided in the manifest(s). If a manifest doesn't provide a
+   * namespace the namespace "default" is used.
+   *
+   * @param manifestFile The manifest file to be modified
+   * @param namespace The namespace that the user specified in configuration.
+   * @return The namespace to be used in the kubectl command.
+   * @throws InterruptedException If an error occurred while reading/writing the manifest file.
+   * @throws IOException If an error occurred while parsing/dumping YAML
+   * @throws IllegalArgumentException If the namespaces of the manifest(s) and the specified
+   *     namespace are inconsistent.
+   */
+  @VisibleForTesting
+  static String addNamespace(FilePath manifestFile, String namespace)
```

Comment:
In order to deploy with a particular namespace when the objects could potentially have different namespaces specified, this is required.

Reply:
Looks like @viglesiasce agrees that it's overstepping, and we'll be okay with letting kubectl apply fail in the non-matching case. I'll need to rework this so that we don't change the users' manifests.